### PR TITLE
add pushover as notification method

### PR DESCRIPTION
**Pull request by antonlindstrom**
This adds https://pushover.net/ as a way of notification.

I'm not sure if this is entirely useful for everybody but I saw myself using it so I thought I'd contribute.

Thanks!

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -56,6 +56,14 @@ func (c Config) Validate() error {
 				return fmt.Errorf("Missing email address in email notification config: %s", proto.MarshalTextString(ec))
 			}
 		}
+		for _, ec := range nc.PushoverConfig {
+			if ec.Token == nil {
+				return fmt.Errorf("Missing token in Pushover notification config: %s", proto.MarshalTextString(ec))
+			}
+			if ec.UserKey == nil {
+				return fmt.Errorf("Missing userkey in Pushover notification config: %s", proto.MarshalTextString(ec))
```
Comment (juliusv) [inline config/config.go]: I'd say "userkey" -> "user key" in a message like this.

File: manager/notifier.go [hunk 2]
```diff
@@ -190,6 +191,15 @@ func (n *notifier) sendEmailNotification(email string, a *Alert) error {
 	return writeEmailBody(wc, a)
 }
 
+func (n *notifier) sendPushoverNotification(token, userKey string, a *Alert) error {
+	po, err := pushover.NewPushover(token, userKey)
+	if err != nil {
+		return err
+	}
+
+	return po.Message("Alertmanager " + a.Description)
```
Comment (juliusv) [inline manager/notifier.go]: Hmm, looking at https://pushover.net/api#messages, pushover seems to support titles for messages, which would be great to put the alert Summary in. But I guess the problem is that the client library so far doesn't support setting a title?

**Comment by juliusv** [comment]
Thanks, looks great otherwise! :+1: 

(yeah, we should add tests both for this and PagerDuty soon...)

**Comment by juliusv** [comment]
And yeah, I think in general, we want to support quite a few notification mechanisms, so this is very useful. If it gets too much, we'll split the code up and make it modular/pluggable...

File: manager/notifier.go (hunk 2 above)
Comment (juliusv) [inline manager/notifier.go]: Actually, the client library does support setting titles:

http://godoc.org/github.com/thorduri/pushover#Pushover.Push

So how about:

```go
_, _, err := po.Push(&po.Message{
	Title: a.Summary,
	Message: a.Description,
})
return err
```

(there are potentially other interesting fields in http://godoc.org/github.com/thorduri/pushover#Message)

**Comment by antonlindstrom** [comment]
@juliusv Thanks for the awesome feedback! I've updated the code based on your comments.

Thanks!

File: README.md [hunk 3]
```diff
@@ -9,7 +9,9 @@ following aspects:
 * inhibiting alerts based on alert dependencies
 * aggregating alerts by labelset
 * handling notification repeats
-* sending alert notifications via external services (currently email or [PagerDuty](http://www.pagerduty.com/))
+* sending alert notifications via external services (currently email,
+[PagerDuty](http://www.pagerduty.com/) or
+[Pushover](https://www.pushover.net/))
```
Comment (juliusv) [inline README.md]: Sorry, just noticed this now: There's one closing parentheses too many.

Comment (antonlindstrom) [inline README.md]: I'm not sure if I understand correctly, the last of the parentheses are ending the one between `..services (currently email..` on row 12. Is that the one you mean?

Here's an image on how the markup renders:

screen shot 2015-02-17 at 12 59 04

Thanks!

Comment (juliusv) [inline README.md]: Oh! Sorry. Due to how it got displayed visually by GitHub in the side-by-side diff variant, I totally missed that. All good.

**Comment by juliusv** [comment]
Thanks a bunch!

**Pull request merged**